app/scripts/views/settings.js

```diff
@@ -3,78 +3,223 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 define([
+  'jquery',
+  'modal',
   'cocktail',
+  'lib/promise',
   'lib/session',
-  'views/form',
   'views/base',
   'views/mixins/avatar-mixin',
+  'views/settings/avatar',
+  'views/settings/avatar_change',
+  'views/settings/avatar_crop',
+  'views/settings/avatar_camera',
+  'views/settings/avatar_gravatar',
+  'views/settings/gravatar_permissions',
+  'views/settings/communication_preferences',
+  'views/change_password',
+  'views/settings/display_name',
+  'views/delete_account',
   'views/mixins/settings-mixin',
+  'views/decorators/allow_only_one_submit',
   'stache!templates/settings'
 ],
-function (Cocktail, Session, FormView, BaseView, AvatarMixin,
-  SettingsMixin, Template) {
+function ($, modal, Cocktail, p, Session, BaseView, AvatarMixin,
+  AvatarView, AvatarChangeView, AvatarCropView, AvatarCameraView, GravatarView,
+  GravatarPermissionsView, CommunicationPreferencesView, ChangePasswordView,
+  DisplayNameView, DeleteAccountView, SettingsMixin, allowOnlyOneSubmit,
+  Template) {
   'use strict';
 
   var t = BaseView.t;
 
-  var View = FormView.extend({
+  var PANEL_VIEWS = [
+    AvatarView,
+    DisplayNameView,
+    CommunicationPreferencesView,
+    ChangePasswordView,
+    DeleteAccountView,
+    AvatarChangeView,
+    AvatarCropView,
+    AvatarCameraView,
+    GravatarView,
+    GravatarPermissionsView
+  ];
+
+  function renderView(view) {
+    return view.render()
+      .then(function (shown) {
+        if (! shown) {
+          view.destroy(true);
+          return;
+        }
+        view.afterVisible();
+
+        return view;
+      });
+  }
+
+  var View = BaseView.extend({
     template: Template,
     className: 'settings',
 
+    FADE_OUT_SETTINGS: 250,
+
     initialize: function (options) {
       options = options || {};
 
       this._able = options.able;
+      this._subViewToShow = options.subView;
+      this._notifications = options.notifications;
+      this._panelViews = options.panelViews || PANEL_VIEWS;
+
+      this.on('navigate-from-subview', this._onNavigateFromSubview.bind(this));
```

Comment:
I find the view listening to a message on itself a bit strange, usually an object listens for messages on another object. The router triggers this message. A view already has a reference to the router, a more natural way of doing this would be to listen for a message on the router, and have the router trigger the message on itself. Is the router an EventEmitter? If not, no harm it making it one.

Also, `navigate-from-subview` is declared as a constant in router.js, perhaps it should be here too, or a common constant set as a static member of the constructor of one of the modules could be shared between the two.